### PR TITLE
[Issue #73]: Dynamic sitemap for pa11y-ci

### LINKED ISSUE #73: [Task]: Have pa11y ci run against a dynamic sitemap
### Summary

Other projects (like jekyll)  generate a dynamic sitemap - that will update routes as they are added to the frontend.

Some possible options:
- https://nextjs.org/docs/app/api-reference/functions/generate-sitemaps
- https://nextjs.org/learn-pages-router/seo/crawling-and-indexing/xml-sitemaps

Due to 

### Acceptance criteria

- Pa11y ci runs against a sitemap

### PR DESCRIPTION
**Pull request by rylew1**
## Summary
Fixes #73 

## Changes proposed
- Generate dynamic sitemap with `/app/sitemap.ts` (next convention)
- Split pa11y config into `pa11y-desktop` and `pa11y-mobile`

Context
===========

sitemap at localhost:3000/sitemap.xml:
image

### CODE REVIEW

File: frontend/src/middleware.ts [hunk 1]
```diff
@@ -19,7 +19,7 @@ export const config = {
      * - _next/image (image optimization files)
      * - images (static files in public/images/ directory)
      */
-    "/((?!api|_next/static|_next/image|public|img|uswds|images|robots.txt|site.webmanifest).*)",
+    "/((?!api|_next/static|_next/image|sitemap|public|img|uswds|images|robots.txt|site.webmanifest).*)",
```
Comment (rylew1) [inline frontend/src/middleware.ts:22]: required to get to sitemap.xml route

File: frontend/src/app/[locale]/health/page.tsx [hunk 2]
```diff
@@ -1,3 +1,10 @@
 export default function Health() {
-  return <>healthy</>;
+  return (
+    <>
+      <head>
+        <title>Health Check</title>
+      </head>
+      <div>healthy</div>
+    </>
```
Comment (rylew1) [inline frontend/src/app/[locale]/health/page.tsx:8]: was getting flagged as an a11y issue, need a title for every page. We also can't use `next/head` / `<Head>` - need to use the default html one

File: frontend/src/app/sitemap.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { MetadataRoute } from "next";
+import { getNextRoutes } from "../utils/getRoutes";
+
+export default function sitemap(): MetadataRoute.Sitemap {
+  const routes = getNextRoutes("./src/app");
+
+  const baseUrl = process.env.NEXT_PUBLIC_BASE_URL || "http://localhost:3000";
+  const sitemap: MetadataRoute.Sitemap = routes.map((route) => ({
+    url: `${baseUrl}${route || ""}`,
```
Comment (acouch) [inline frontend/src/app/sitemap.ts:9]: Just noting we will want to add `${locale}` once we have spanish translations.

File: frontend/src/utils/getRoutes.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
+import fs from "node:fs";
+import path from "node:path";
+
+// Helper function to list all paths recursively
+function listPaths(dir: string): string[] {
```
Comment (acouch) [inline frontend/src/utils/getRoutes.ts]: This is a nice dynamic solution instead of hard coding. We should add a test for the files in this function.

**Review by btabaska — APPROVED** [review]

**Pull request merged**